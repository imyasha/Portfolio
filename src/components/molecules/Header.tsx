import React from "react"
import { Link } from "gatsby"
import ToggleMode from "../atoms/ToggleMode"
import styled from "styled-components"

import Highlight from "../atoms/Highlight"
import Menu from "../atoms/Menu"

const Head = styled.header`
  padding: 1rem 3%;
  display: flex;
  justify-content: space-between;

  h1 {
    margin: 0;
    line-height: 1;
  }
`

const Navigation = styled.nav`
  display: flex;
  flex-direction: row;
  justify-content: center;
  height: 1em;
  line-height: 1;

  button {
    background: transparent;
    color: inherit;
    border: none;
  }
`

const Header = () => {
  return (
    <Head>
      <h1>
        <Link to="/">
          &lt;Jacob Herper <Highlight>/</Highlight>&gt;
        </Link>
      </h1>
      <Navigation>
        <Menu>
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/about/">About</Link>
          </li>
          <li>
            <Link to="/portfolio/">Portfolio</Link>
          </li>
          <li>
            <Link to="/contact/">Contact</Link>
          </li>
          <li>
            <ToggleMode />
          </li>
        </Menu>
      </Navigation>
    </Head>
  )
}

export default Header
