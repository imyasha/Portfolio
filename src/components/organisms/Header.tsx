import React, { useRef } from 'react';
import { Link } from 'gatsby';
import { motion, useCycle } from 'framer-motion';
import styled from 'styled-components';

import { useDimensions } from 'Hooks/useDimensions';
import { MenuToggle } from 'Atoms/MenuToggle';
import { Navigation as Nav } from 'Molecules/Navigation';
import Logo from 'Atoms/Logo';

// import NavigationContext from "Context/navigation.context"
import ToggleMode from 'Atoms/ToggleMode';

const sidebar = {
  open: (height = 1080) => ({
    clipPath: `circle(${height * 2 + 200}px at 40px 40px)`,
    transition: {
      type: 'spring',
      stiffness: 20,
      restDelta: 2,
    },
  }),
  closed: {
    clipPath: 'circle(30px at 40px 40px)',
    transition: {
      delay: 0.5,
      type: 'spring',
      stiffness: 400,
      damping: 40,
    },
  },
};

const Header = () => {
  // const MenuToggle = useContext(NavigationContext)
  const [isOpen, toggleOpen] = useCycle(false, true);
  const containerRef = useRef(null);
  const { height } = useDimensions(containerRef);

  return (
    <Head>
      <Toggler
        initial={false}
        animate={isOpen ? 'open' : 'closed'}
        custom={height}
        ref={containerRef}
      >
        <Background className="menu-background" variants={sidebar} />
        <Nav toggle={() => toggleOpen()} />
        <MenuToggle toggle={() => toggleOpen()} />
      </Toggler>
      <Link to="/" data-testid="logo">
        <Logo />
      </Link>
      <ThemeToggle data-testid="themeToggle" />
    </Head>
  );
};

const Background = styled(motion.div).attrs(() => ({
  initial: 'closed',
  sidebar,
}))`
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  width: 300px;
`;

const Toggler = styled(motion.nav)`
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  width: auto;
  z-index: 1;
`;

const Head = styled.header`
  padding: 1rem 3%;
  display: flex;
  justify-content: center;

  h1 {
    margin: 0;
    line-height: 1;

    @media screen and (max-width: 375px) {
      font-size: 1.25rem;
    }

    @media screen and (max-width: 768px) {
      font-size: 1.75rem;
    }
  }

  @media print {
    display: none;
  }
`;

const ThemeToggle = styled(ToggleMode)`
  position: absolute;
  right: 0;
`;

export default Header;
