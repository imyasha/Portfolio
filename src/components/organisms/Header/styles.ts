import styled from 'styled-components';

const Flex = styled.div`
	display: flex;
	justify-content: space-between;
	align-items: center;

	a {
		color: ${({ theme }) => theme.text};
		text-decoration: none;
		transition: 0.2s;

		&:hover {
			color: ${({ theme }) => theme.headline};
		}
	}
`;

const Desktop = styled.div`
	display: none;

	@media screen and (min-width: 768px) {
		display: block;
	}
`;

const Mobile = styled.div`
	display: block;

	@media screen and (min-width: 768px) {
		display: none;
	}
`;

export { Desktop, Mobile, Flex };
