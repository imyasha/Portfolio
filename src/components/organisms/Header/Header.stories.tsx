import { Header } from './Header';

export default {
	title: 'organisms/Header',
	component: Header,
};

export const HeaderStory = () => <Header />;

HeaderStory.storyName = 'Header';
HeaderStory.args = {};
