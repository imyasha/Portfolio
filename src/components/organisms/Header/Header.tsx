import { FC } from 'react';
import Link from 'next/link';
import { Desktop, Flex, Mobile } from './styles';

import { Container } from 'Atoms/Container';
import { MobileNavigation } from 'Molecules/MobileNavigation';
import { Navigation } from 'Molecules/Navigation';

const Header: FC = () => {
	return (
		<header>
			<Container>
				<Flex>
					<h1>
						<Link href="/">
							<a>Jacob Herper</a>
						</Link>
					</h1>
					<Desktop>
						<Navigation />
					</Desktop>
					<Mobile>
						<MobileNavigation />
					</Mobile>
				</Flex>
			</Container>
		</header>
	);
};

export { Header };
