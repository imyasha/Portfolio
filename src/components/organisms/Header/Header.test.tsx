import { Header } from '../Header';
import { render } from '@testing-library/react';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
	useSelector: jest.fn(),
	useDispatch: () => mockDispatch,
}));

describe('Header', () => {
	it('renders correctly', () => {
		const { container } = render(<Header />);
		expect(container).toMatchSnapshot();
	});
});
