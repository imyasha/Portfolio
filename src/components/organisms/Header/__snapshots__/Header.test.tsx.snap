// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <header>
    <div
      class="styles__StyledContainer-mepagl-0 exxZos"
    >
      <div
        class="styles__Flex-sc-14e6kmf-0 iaPxwS"
      >
        <h1>
          <a
            href="/"
          >
            Jacob Herper
          </a>
        </h1>
        <div
          class="styles__Desktop-sc-14e6kmf-1 grzcZM"
        >
          <nav
            class="styles__StyledNavigation-sc-16epmp-0"
          >
            <ul
              class="styles__List-sc-16epmp-1 gPSCMv"
            >
              <li>
                <a
                  href="/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  href="/about"
                >
                  About
                </a>
              </li>
              <li>
                <a
                  href="/uses"
                >
                  Uses
                </a>
              </li>
              <li>
                <a
                  href="/blog"
                >
                  Blog
                </a>
              </li>
              <li>
                <a
                  href="/portfolio"
                >
                  Portfolio
                </a>
              </li>
              <li
                class="desktop-only"
              >
                <button
                  class="styles__StyledThemeToggle-sc-1dfmkyd-0 iOJryI"
                  title="Switch to light theme"
                >
                  <test-file-stub />
                </button>
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="styles__Mobile-sc-14e6kmf-2 sgtTF"
        >
          <ul
            class="styles__StyledMobileNavigation-sc-14uzdzl-0 crjHjf"
          >
            <li>
              <button
                class="styles__StyledThemeToggle-sc-1dfmkyd-0 iOJryI"
                title="Switch to light theme"
              >
                <test-file-stub />
              </button>
            </li>
            <li>
              <button
                class="styles__Menu-sc-14uzdzl-1 dJJTrF"
              >
                <test-file-stub />
              </button>
            </li>
          </ul>
          <div
            class="styles__Overlay-sc-14uzdzl-2 grizif"
          >
            <nav
              class="styles__StyledNavigation-sc-16epmp-0"
            >
              <ul
                class="styles__List-sc-16epmp-1 gPSCMv"
              >
                <li>
                  <a
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li>
                  <a
                    href="/about"
                  >
                    About
                  </a>
                </li>
                <li>
                  <a
                    href="/uses"
                  >
                    Uses
                  </a>
                </li>
                <li>
                  <a
                    href="/blog"
                  >
                    Blog
                  </a>
                </li>
                <li>
                  <a
                    href="/portfolio"
                  >
                    Portfolio
                  </a>
                </li>
                <li
                  class="desktop-only"
                >
                  <button
                    class="styles__StyledThemeToggle-sc-1dfmkyd-0 iOJryI"
                    title="Switch to light theme"
                  >
                    <test-file-stub />
                  </button>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
